app/assets/javascripts/setup.js

```diff
@@ -17,4 +17,32 @@ $(function() {
       $(this).text('Show');
     }
   });
-});
\ No newline at end of file
+
+  $(document).on('click', '.js-toggle-registry-settings-btn', function() {
+    var targetId = $(this).data('target');
+    var collapsed = $(targetId).attr('aria-expanded') === 'true';
+
+    if (collapsed) {
+      $(this).text('Hide');
+    } else {
+      $(this).text('Show');
+    }
+  });
+
+  $(document).on('click', '.btn-add-registry', function() {
+    var registryForm = $('.registry-form'),
+        currentEntry = $(registryForm).children('.registry-entry:first'),
+        newEntry = $(currentEntry.clone()).appendTo(registryForm);
+
+    newEntry.find('#settings_insecure_registries__docker_registry_name').val('').prop('disabled', false);
+    newEntry.find('#settings_insecure_registries__docker_registry_mirror').prop('checked', false);
+    newEntry.find('#settings_insecure_registries__docker_registry_certificate').val('');
+    registryForm.find('.registry-entry:not(:last) .btn-add-registry')
+        .removeClass('btn-add-registry').addClass('btn-remove-registry')
+        .removeClass('btn-success').addClass('btn-danger')
+        .html('<span class="glyphicon glyphicon-minus"></span>');
+  }).on('click', '.btn-remove-registry', function(e) {
+    console.log($(this).parents('.row .registry-entry:first'));
+    $(this).parents('.row .registry-entry:first').remove();
```

Comment:
Use '.closest' when looking for a parent node.

Reply:
You can probably get rid of the `:first` pseudoclass too.

Reply:
removing the pseudoclass works, but closest didnt, i wrote it like this:

```js
$(this).closest().remove();
```

Reply:
Try `.closest('.registry-entry')`